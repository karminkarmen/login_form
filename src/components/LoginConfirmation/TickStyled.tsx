import styled from 'styled-components';
import { Tick } from '@styled-icons/typicons';
import { innerWidthSize } from '../../constants/themeConsts';

export const TickStyled = styled(Tick)`
    color: ${props => props.theme.colors.main};
    height: ${innerWidthSize / 2}px;
`
export default TickStyled;
