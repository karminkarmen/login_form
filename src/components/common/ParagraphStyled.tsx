import styled from 'styled-components';

const ParagraphStyled = styled.p`
    margin: 0 0 ${props => props.theme.margins.normal} 0;
    max-width: ${props => props.theme.width.innerElementWidth};
    color: ${props => props.theme.colors.main};
    font-size: ${props => props.theme.fontSizes.normal};
    font-family: ${props => props.theme.fontFamilies.lato};
`;

export default ParagraphStyled;