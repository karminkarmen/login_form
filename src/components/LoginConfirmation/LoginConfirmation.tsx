import React from 'react';
import { ThemeProvider } from 'styled-components';
import theme from '../../configs/theme';
import ParagraphStyled from '../common/ParagraphStyled';
import messages from '../../constants/messagesConsts';
import { TickStyled } from './TickStyled';

const LoginConfirmation: React.FC = () => (
    <ThemeProvider theme={theme}>
        <TickStyled />
        <ParagraphStyled bigFontSize textSpace>
            {messages.loggedInConfirmationInfo}
        </ParagraphStyled>
    </ThemeProvider>
);

export default LoginConfirmation;
