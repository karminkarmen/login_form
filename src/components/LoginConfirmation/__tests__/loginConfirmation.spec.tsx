import React from 'react';
import { shallow } from 'enzyme';

import LoginConfirmation from '../LoginConfirmation';
import ParagraphStyled from '../../common/ParagraphStyled';
import messages from '../../../constants/messagesConsts';
import TickStyled from '../TickStyled';

describe('<LoginConfirmation />', () => {

    it('matches snapshot', () => {
        const wrapper = shallow(<LoginConfirmation />);
        expect(wrapper.debug()).toMatchSnapshot();
    });
    
    it('should render styled paragraph', () => {
        const wrapper = shallow(<LoginConfirmation />);

        expect(wrapper.find(ParagraphStyled)).toHaveLength(1);
    });

    it('should render styled paragraph with login confirmation message', () => {
        const wrapper = shallow(<LoginConfirmation />);

        const label = wrapper.find(ParagraphStyled);
        expect(label.text()).toEqual(messages.loggedInConfirmationInfo);
    });

    it('should render styled tick icon', () => {
        const wrapper = shallow(<LoginConfirmation />);

        expect(wrapper.find(TickStyled)).toHaveLength(1);
    });

});
