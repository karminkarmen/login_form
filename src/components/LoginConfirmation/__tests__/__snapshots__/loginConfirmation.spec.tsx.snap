// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginConfirmation /> matches snapshot 1`] = `
"<ThemeProvider theme={{...}}>
  <Styled(Tick) />
  <styled.p bigFontSize={true} textSpace={true}>
    login successful
  </styled.p>
</ThemeProvider>"
`;
